tool/tsh/aws.go

```diff
@@ -0,0 +1,344 @@
+	}
+	key, err := tc.LocalAgent().GetKey(tc.SiteName, client.WithAppCerts{})
+	if err != nil {
+		return tls.Certificate{}, trace.Wrap(err)
+	}
+	cc, ok := key.AppTLSCerts[tc.CurrentAWSCLIApp]
+	if !ok {
+		return tls.Certificate{}, trace.NotFound("please login into AWS Console App 'tsh app login' first")
+	}
+	cert, err := tls.X509KeyPair(cc, key.Priv)
+	if err != nil {
+		return tls.Certificate{}, trace.Wrap(err)
+	}
+	return cert, nil
+}
+
+func validateARNRole(cf *CLIConf, tc *client.TeleportClient, profile *client.ProfileStatus, arnRole string) error {
+	ok := awsarn.IsARN(arnRole)
+
+	validARNs := make(map[string]struct{})
+	for _, roleName := range profile.Roles {
+		role, err := tc.GetRole(cf.Context, roleName)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		roleARNs := role.GetAWSRoleARNs(types.Allow)
+		for _, v := range roleARNs {
+			if v == arnRole {
+				return nil
+			}
+			validARNs[v] = struct{}{}
+		}
+	}
+	printMapKeysAs(validARNs, "Available ARNs")
+	if !ok {
+		// User provided invalid formatted ARN role string, print all available ARN roles for the user and indicate
+		// and indicate about invalid ARN format.
+		return trace.BadParameter("invalid AWS ARN role format: %q", arnRole)
+	}
+	return trace.NotFound("user is not allowed to use selected AWS ARN role: %q.", arnRole)
+}
+
+func printMapKeysAs(m map[string]struct{}, columnName string) {
+	if len(m) == 0 {
+		return
+	}
+	t := asciitable.MakeTable([]string{columnName})
+	for k := range m {
+		t.AddRow([]string{k})
+	}
+	fmt.Println(t.AsBuffer().String())
+
+}
+
+// findARNBasedOnRoleName tries to match roleName parameter with allowed user ARNs obtained from the Teleport API based on
+// user roles profile. If there is a match the IAM role is created based on accountID and roleName fields.
+func findARNBasedOnRoleName(cf *CLIConf, tc *client.TeleportClient, profile *client.ProfileStatus, accountID, roleName string) (string, error) {
+	validRolesName := make(map[string]struct{})
+	for _, v := range profile.Roles {
+		role, err := tc.GetRole(cf.Context, v)
+		if err != nil {
+			return "", trace.Wrap(err)
+		}
+		for _, v := range role.GetAWSRoleARNs(types.Allow) {
+			arn, err := awsarn.Parse(v)
+			if err != nil {
+				return "", trace.Wrap(err)
+			}
+
+			// Example of the ANR Resource: 'role/EC2FullAccess'
+			parts := strings.Split(arn.Resource, "/")
+			if len(parts) != 2 {
+				continue
+			}
+			if parts[0] != "role" {
+				continue
+			}
+
+			if arn.AccountID == accountID {
+				validRolesName[parts[1]] = struct{}{}
+			}
+			if arn.AccountID == accountID && parts[1] == roleName {
+				return arn.String(), nil
+			}
+		}
+	}
+	if len(validRolesName) != 0 {
+		printMapKeysAs(validRolesName, "Available Roles")
+	}
+	return "", trace.NotFound("failed to find role ARN based on AWSAccountID(%q) and RoleName(%q)", accountID, roleName)
+}
+
+func setFakeAWSEnvCredentials(keyID, accessKey string) error {
```

Comment:
Nit: I would call parameters `accessKey` and `secretKey` to keep AWS terminology.